docs/reference/mapping/params/collapsed.asciidoc

```diff
@@ -0,0 +1,102 @@
+[[collapsed]]
+=== `collapsed`
+
+When indexing a document or updating mappings, Elasticsearch accepts field that contain dots in their names.
+which get expanded to their corresponding object structure. For instance, the  field `metrics.time.max`
+is mapped as a `max` leaf field with a parent `time` object, belonging to its parent `metrics` object.
+
+The described default behaviour is reasonable for most scenarios, but causes problems in certain situations
+where for instance a field `metrics.time` holds a value too, which is common when indexing metrics data.
+A document holding a value for both `metrics.time.max` and `metrics.time` gets rejected given that `time`
+would need to be a leaf field to hold a value as well as an object to hold the `max` sub-field.
+
+The `collapsed` setting, which can be applied only to the top-level mapping definition and
+to <<object,`object`>> fields, makes it possible to store metrics where field names contain dots and share
+common prefixes. From the example above, if the object container `metrics` is collapsed, it can hold values
+for both `time` and `time.max` directly without the need for any intermediate object, as dots in field
+names are preserved.
```

Comment:
Maybe worth saying here that you can't add a document with an object structure underneath a collapsed field?

Reply:
yep i thought I had that but not here I see, good point